scipp/qens-reduction.ipynb

```diff
@@ -0,0 +1,409 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from scippneutron.conversion.graph.beamline import beamline\n",
+    "from scippneutron.conversion.tof import energy_transfer_indirect_from_tof\n",
+    "\n",
+    "graph = {\n",
+    "    **beamline(scatter=True),\n",
+    "    \"L2\": backscattered_l2,\n",
+    "    \"final_wavelength\": wavelength_from_analyzer,\n",
+    "    \"final_energy\": final_energy,\n",
+    "    \"energy_transfer\": energy_transfer_indirect_from_tof,\n",
+    "}\n",
+    "del graph[\"two_theta\"]\n",
+    "del graph[\"scattered_beam\"]\n",
+    "del graph[\"Ltotal\"]\n",
+    "sc.show_graph(graph, simplified=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fa3bfc7d-9716-4753-a38d-9cdc95c77c6e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def correct_tof(data):\n",
+    "    data = data.copy()\n",
+    "    # TODO according to Mads' script\n",
+    "    #  Instrument focuses at center of pulse, 2.86 ms / 2\n",
+    "    shift = sc.scalar(0.5 * 2.86, unit=\"ms\")\n",
+    "    if \"tof\" in data.coords:\n",
+    "        data.coords[\"tof\"] -= shift.to(unit=data.coords[\"tof\"].unit)\n",
+    "    if data.bins and \"tof\" in data.bins.coords:\n",
+    "        data.bins.coords[\"tof\"] -= shift.to(unit=data.bins.coords[\"tof\"].bins.unit)\n",
+    "    return data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c3c3175a-8227-4da4-938d-7a0c6d01200e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def to_energy_transfer(data):\n",
+    "    data = correct_tof(data)\n",
+    "    return data.transform_coords(\"energy_transfer\", graph=graph)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "112d54f7-6a22-48d1-a9a9-4ab91b206903",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "unmasked = to_energy_transfer(events)\n",
+    "unmasked"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7bd44f00-ee45-4b85-968a-a49561915dad",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "masked = to_energy_transfer(binned_for_mask)\n",
+    "masked"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bf1fd84e-8de7-4830-8e0f-db43e109ccdf",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "masked_hist = masked.bins.concat().hist(energy_transfer=100)\n",
+    "unmasked_hist = unmasked.hist(energy_transfer=100)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3f6a1af2-7992-4199-a973-8b2f34d51a07",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "utils.add_variances(masked_hist)\n",
+    "utils.add_variances(unmasked_hist)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "80945a3e-ccbe-4643-98cf-2a42b9b651af",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pp.plot(\n",
```

Comment:
Since they will do the same to all 3 files (once we have 3 files in the notebook), does it make sense to teach them about `DataGroup`, and convert everything in a single call? (after they have done it for one file)

Reply:
We could. But this adds one more thing to learn. Currently, I'm leaning towards not telling them how to do it and just letting them figure it out and use a loop or (hopefully not) copy-paste.